modules/core/src/main/scala/doobie/util/fragments.scala

```diff
@@ -12,6 +12,14 @@ import cats.syntax.all._
 /** Module of `Fragment` constructors. */
 object fragments {
 
+  /** Returns `VALUES (fs0, fs1, ...)`. */
+  def tableValues[F[_]: Reducible, A: util.Put](fs: F[A]): Fragment =
```

Comment:
* If you use `(implicit w: Write[A])` instead and call `w.toFragment`, you won't need the method below. This will work for single column values & product types too.
* I might be wrong but the extra paren after `VALUES` seems unnecessary. `SELECT * FROM (VALUES (3),(2)) a;` seems to work fine for me in Postgres. (MySQL has very different syntax for table values. @tpolecat do we tend to only put standard SQL in this file?)
* Is there an issue calling this `values` following the the names of other functions in this file?

Reply:
- I agree that `Write` can be derived for tuple/product types, but is it possible to derive it for variable-length data structures like `NonEmptyList` as well which is what my initial point was about?
- Agree, will fix.

Reply:
No, Write doesn't get derived for List/NEL so it's fine.

Reply:
So why don't we need the method? With the current implementation it's possible to write the following:

```
def query(ids: NonEmptyList[Int]) = 
    sql"select * from (${fragments.values(ids)}) s(a) join some_tbl on a=some_tbl.id"
```

`fragments.values` here provides the `VALUES (?), (?), (?)` which is supposed to be written by hand otherwise.

Reply:
Hi, @jatcwang . Are there any updates from your side?

Reply:
@dmitriibundin Hi sry missed your msg. I'll take a look later

Reply:
Looking through the thread it seems like there might be some miscommunication of what I meant 
and why I suggetsted using `Write`.

Basically I'm trying to generalize this to allow any F[A] as long as `A` has a `Write` instance

```
case clsas MyObj(a: Int, s: String, d: Double)

val myObjs: NonEmptyList[MyObj] = ...

sql"SELECT * from (${fragments.values(myObj)}) s(a) join ..."
```

So the signature of `values` that I'm suggesting is this

```
def values[F[_]: Reducible, A](fs: F[A])(implicit w: Write[A]): Fragment = ...
```

This will work whether `A` is `Int` (single column) or a case class / tuple. It's more general :)

Implementation-wise you can use `w.toFragment`. Let me know if you need more help.

Reply:
Hi @jatcwang, thanks for the detailed clarification. Updated the PR according to your latest comment.